crates/task-impls/src/helpers.rs

```diff
@@ -60,6 +60,29 @@ pub async fn calculate_vid_disperse<TYPES: NodeType>(
     VidDisperse::from_membership(view, vid_disperse, membership.as_ref())
 }
 
+/// Calculate the vid disperse information from the payload given a view and membership, and precompute data from builder
+///
+/// # Panics
+/// Panics if the VID calculation fails, this should not happen.
+pub async fn calculate_vid_disperse_using_precompute_data<TYPES: NodeType>(
+    txns: Vec<u8>,
+    membership: Arc<TYPES::Membership>,
+    view: TYPES::Time,
+    pre_compute_data: VidPrecomputeData,
+) -> VidDisperse<TYPES> {
+    let num_nodes = membership.total_nodes();
+    let vid_disperse = spawn_blocking(move || {
```

Comment:
Can we just move the #[allow(panic)] to the top of the function?

Reply:
https://github.com/EspressoSystems/HotShot/pull/2956/commits/8d15da890e00f3ab93edcd880b5fb1160b95768d